### DISCUSSION TITLE
Why aren't fixtures co-located with tests?

### DISCUSSION THREAD

**Discussion by supafreq** [PaulRBerg/hardhat-template#54] [Q&A]
I wanted to thank you for pointing me toward your Hifi repo where you implement this pattern at scale- it really helped answer most of my questions. Something I'm still wondering is, is there any reason why your fixtures aren't co-located with their corresponding tests? It would make sense to have them in the `shared` directory if they were, well, shared, but as it stands right now, they aren't. (I'm aware this is a question directed toward the Hifi repo, but since it's implementing the pattern laid out here, I thought this was the place to ask.)

**Comment by PaulRBerg** [comment]
It's a good question. It took me a minute to remember why I did it this way.

The fixtures are "shared" because they are all aggregated in one place, in that they all need the same package imported and such. I agree though that this is misleading - it would have made more sense if each individual fixture was used in multiple locations. I might do a refactor someday.
